src/Magnum/GL/AbstractShaderProgram.h

```diff
@@ -1446,13 +1447,36 @@ class MAGNUM_GL_EXPORT AbstractShaderProgram: public AbstractObject {
         /**
          * @brief Link the shader
          *
-         * Links single shader. If possible, prefer to link multiple shaders
-         * at once using @ref link(std::initializer_list<Containers::Reference<AbstractShaderProgram>>)
+         * Calls @ref submitLink(), then @ref checkLink().
+         * If possible, prefer to link multiple shaders at once using
+         * @ref link(std::initializer_list<Containers::Reference<AbstractShaderProgram>>)
          * for improved performance, see its documentation for more
          * information.
          */
         bool link();
 
+        /**
+         * @brief Submit for linking
+         *
+         * The attached shaders must be compiled with @ref Shader::compile() before linking.
```

Comment:
I get the sentence copied from the `link()` above, but now it should probably mention that `submitCompile()` is also enough? :)